projects/com.oracle.truffle.llvm.nodes/src/com/oracle/truffle/llvm/nodes/memory/LLVMAllocInstruction.java

```diff
@@ -88,13 +89,28 @@ public int getLength() {
             return offsets.length;
         }
 
+    }
+
+    public abstract static class LLVMAllocaConstInstruction extends LLVMAllocConstInstruction {
+
         @Specialization
         protected LLVMNativePointer doOp(VirtualFrame frame,
                         @Cached("getLLVMMemory()") LLVMMemory memory) {
             return LLVMNativePointer.create(LLVMStack.allocateStackMemory(frame, memory, getStackPointerSlot(), getSize(), getAlignment()));
         }
     }
 
+    @NodeField(type = UniqueSlot.class, name = "uniqueSlot")
+    public abstract static class LLVMUniqueAllocConstInstruction extends LLVMAllocConstInstruction {
```

Comment:
As far as I understand your PR, this is not actually "allocating" anything, but returning the pointer to a pre-allocated stack location, right?
In that case, the node should probably not be called `...Alloc...Instruction`, but rather `LLVMGetUniqueStackPointerInstruction` or something like that.

Reply:
> As far as I understand your PR, this is not actually "allocating" anything, but returning the pointer to a pre-allocated stack location, right?

Exactly.

> In that case, the node should probably not be called ...Alloc...Instruction, but rather LLVMGetUniqueStackPointerInstruction or something like that.

I agree that `LLVMUniqueAllocConstInstruction` does not fit well. The problem is, since the `LLVMUniqueAllocConstInstruction` also IS an `LLVMAllocInstruction`, one could argue that consequently the superclasses `LLVMAllocInstruction` and `LLVMAllocConstInstruction` need to be renamed for a similar reason. Those classes provide functionality for stack allocated memory (`getSize()`, `getAlignment()`, `getOffsets()` etc.) which is independent from whether it is applied to newly allocated memory or pre-allocated memory (that is why I made my node part of this type hierarchy). This ambivalence between "merely getting a pointer to memory" and "allocating memory" should be reflected by the name then. I thought about expressing this using words like "claim" (close to "allocate", but could also be understood in the sense of "claiming a right on something which already belongs to the one claiming" ) or "obtain" (close to "get", but expresses that there is potentially a bit of additional effort - allocation in our context - involved).

How about:
`LLVMAllocInstruction` -> `LLVMClaimStackSpaceInstruction`
`LLVMAllocConstInstruction` -> `LLVMClaimStackForConstInstruction`
`LLVMUniqueAllocConstInstruction` -> `LLVMGetUniqueStackPointerInstruction`

Things like `createUniqueAlloc` would also need to be renamed accordingly.

Reply:
I actually like "get" better than "claim" or "obtain" for the base class. It's simply getting a pointer to some stack memory. For some subclasses it's newly allocated memory, for some subclasses it's preallocated.

How about:
`LLVMAllocInstruction` -> `LLVMGetStackSpaceInstruction`
leave `LLVMAllocaInstruction` and `LLVMAllocaConstInstruction` (with the "a", they represent the `alloca` bitcode and `alloca` C function respectively)
`LLVMUniqueAllocConstInstruction` -> `LLVMGetUniqueStackSpaceInstruction`

Reply:
Ok. I pushed changes implementing the naming scheme you suggested. The commit also includes the requested documentation work on `UniquesRegion` and the removal of code duplication (`setMSB` -> `Integer.highestOneBit`).

Reply:
Thanks, lgtm.